### PR TITLE
Feature/quick start notification

### PR DESCRIPTION
**Pull request by khaykov**
Part of #7707 

**Do not merge this PR untill [Flux-C counterpart](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/878) is merged and reference to correct commit are updated.**

This PR adds a Quick Start completion notification.
Notification should not be fired when you skip QuickStart.

In case, when Quick Start was completed offline, or notification request somehow did not reached the server, we will try to request notification when opening app again, changing site or navigating to main activity again.

This PR also adds couple of minor changes to the code:
- We are using separate flag to indicate that quick start was completed (instead of `isTaskShown`)
- When individual quick start task prompt was shown maximum number of times it will not start apearing again after you complete a task naturally (when not prompted or not by selecting it from Quick Start list)

To test:

1 - Complete a Quick Start normally.
2 - Make sure you get a notification.

1 - Complete a Quick Start while offline.
2 - Make sure you get a notification after you navigate back to main activity with connection to internet.

1 - Complete a Quick Start while offline.
2 - Switch to some other site.
3 - Connect to internet
4 - Switch back to original site with completed quick start.
5 - Make sure you get a notification.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListFragment.java [hunk 1]
```diff
@@ -49,6 +50,7 @@
 
     @Inject AccountStore mAccountStore;
     @Inject QuickStartStore mQuickStartStore;
+    @Inject Dispatcher mDispatcher;
```
Comment (theck13) [inline WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListFragment.java]: How does this `Dispatcher` work if we don't register and unregister it?

File: WordPress/src/main/java/org/wordpress/android/util/QuickStartUtils.kt [hunk 2]
```diff
@@ -156,15 +158,32 @@ class QuickStartUtils {
...
+        @JvmStatic
+        fun completeTask(
+            quickStartStore: QuickStartStore,
+            task: QuickStartTask,
+            dispatcher: Dispatcher,
+            site: SiteModel
+        ) {
+            val siteId = site.id.toLong()
+
+            if (quickStartStore.getQuickStartCompleted(siteId) || isEveryQuickStartTaskDone(quickStartStore)) {
```
[the remark this answers is not in the document]
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/util/QuickStartUtils.kt]: `if` condition here is to check if the quick start was completed in one way or another.
If `quickStartStore.getQuickStartCompleted(siteId)` can be `false` and `isEveryQuickStartTaskDone(quickStartStore)` `true` when you completed all the tasks but have not "finilized" the quick start by checking the task list on more last time (when orange dot is visible).

Additionally, `quickStartStore.getQuickStartCompleted(siteId)` can be `true` while `isEveryQuickStartTaskDone(quickStartStore)` is `false` when you skipped quick start.

File: WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListFragment.java (hunk 1 above)
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListFragment.java]: You only need to register a dispatcher if you want to receive events - you can dispatch them without registering it.

Comment (theck13) [reply WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListFragment.java]: Ahh, okay.  That works for me!

File: WordPress/src/main/java/org/wordpress/android/util/QuickStartUtils.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (theck13) [reply WordPress/src/main/java/org/wordpress/android/util/QuickStartUtils.kt]: That makes sense.  Thanks for the explanation!

**Pull request merged**